Source/WebCore/Modules/WebGPU/GPUBuffer.cpp

```diff
@@ -47,12 +53,15 @@ void GPUBuffer::mapAsync(GPUMapModeFlags mode, std::optional<GPUSize64> offset,
 
 Ref<JSC::ArrayBuffer> GPUBuffer::getMappedRange(std::optional<GPUSize64> offset, std::optional<GPUSize64> size)
 {
-    auto mappedRange = m_backing->getMappedRange(offset.value_or(0), size);
-    return ArrayBuffer::create(mappedRange.source, mappedRange.byteLength);
+    m_mappedRange = m_backing->getMappedRange(offset.value_or(0), size);
+    m_arrayBuffer = ArrayBuffer::create(m_mappedRange.source, m_mappedRange.byteLength);
+    return m_arrayBuffer;
 }
 
 void GPUBuffer::unmap()
 {
+    memcpy(m_mappedRange.source, m_arrayBuffer->data(), m_mappedRange.byteLength);
```

Comment:
Should the JS array buffer be reset after the copy?

Reply:
Yes I will address that